mineral-ui/mineral-ui
# [Popover][Dropdown]: Improve a11y

### Description

This PR extracts the a11y and refactoring improvements from https://github.com/mineral-ui/mineral-ui/pull/414 as the portal related updates are currently blocked by an issue in Popperjs. 

### Motivation and context

I am not sure how long it may take to get a new version of Popper and I would like to get these  improvements into the codebase sooner rather than later.

connects #395 

### Screenshots, videos, or demo, if appropriate

https://395-popover-misc--mineral-ui.netlify.com/

### How to test

1. Mostly just try out Dropdown menus in voiceover.  Focus, nav, and announcements are improved.

_This code has already been reviewed and approved in https://github.com/mineral-ui/mineral-ui/pull/414, so as long as I extracted it cleanly from that PR, we should be good to go._

### Types of changes

- Other (provide details below)

Little bit of everything - improved a11y, refactoring, updating tests

### Checklist
